spec/generators/data_migration/data_migration_generator_spec.rb

```diff
@@ -58,9 +62,49 @@
           'data_migration.rb', data_migrations_file_path
         )
 
+        expect(subject).not_to receive(:template)
+
         subject.create_data_migration
       end
     end
+
+    context 'when test support is enabled' do
+      let(:rails_root) { 'dummy-app' }
+
+      before do
+        DataMigrate.config.data_migrations_path = 'db/data/'
+        DataMigrate.config.test_support_enabled = true
+
+        allow(Rails).to receive(:root).and_return(Pathname.new(rails_root))
+        expect(DataMigrate::Helpers::InferTestSuiteType).to receive(:new).and_return(infer_double)
+      end
+
+      context 'and the test suite is RSpec' do
+        let(:infer_double) { instance_double(DataMigrate::Helpers::InferTestSuiteType, call: :rspec) }
+
+        it 'creates a spec file' do
+          expect(subject).to receive(:template).with(
+            'data_migration_spec.rb',
+            "#{rails_root}/spec/db/data/#{migration_name}_spec.rb"
+          )
+
+          subject.create_data_migration
+        end
+      end
+
+      context 'and the test suite is Minitest' do
+        let(:infer_double) { instance_double(DataMigrate::Helpers::InferTestSuiteType, call: :minitest) }
+
+        it 'creates a test file' do
+          expect(subject).to receive(:template).with(
+            'data_migration_test.rb',
+            "#{rails_root}/test/db/data/#{migration_name}_test.rb"
+          )
+
+          subject.create_data_migration
+        end
+      end
+    end
```

Comment:
these specs seem to create the data migration file here, but doesn't clean it up from the local repo once specs are done running.
```
    when test support is enabled
      and the test suite is RSpec
   identical  db/data/20241225214448_my_migration.rb
        creates a spec file
      and the test suite is Minitest
   identical  db/data/20241225214448_my_migration.rb
```

I'd guess this is coming from a part of the code you didn't mean to test in the first place. I wonder if there's a clean post suite hook we could write to remove any file actually created as part of the specs.